chalk/core.py

```diff
@@ -69,10 +70,19 @@ def _style(self, style: Style) -> Diagram:
     def compose(
         self, envelope: Envelope, other: Optional[Diagram] = None
     ) -> Diagram:
-        other = other if other is not None else Empty()
+        if other is None and isinstance(self, Compose):
+            return Compose(envelope, self.diagrams)
+        if other is None and isinstance(self, Compose):
+            return Compose(envelope, [self])
+        
+        other = other if other is not None else Empty() 
+        if isinstance(self, Empty):
+            return other
+        if isinstance(other, Empty):
+            return self
```

Comment:
Shouldn't we use the given envelope for the results returned here and at line 80?